salt/modules/suse_ip.py

```diff
@@ -0,0 +1,1174 @@
+    if retain_settings:
+        for opt in current:
+            nopt = opt
+            if opt == "netconfig_dns_static_servers":
+                nopt = "dns"
+                result[nopt] = current[opt].split()
+            elif opt == "netconfig_dns_static_searchlist":
+                nopt = "dns_search"
+                result[nopt] = current[opt].split()
+            elif opt.startswith("netconfig_") and opt not in (
+                "netconfig_modules_order",
+                "netconfig_verbose",
+                "netconfig_force_replace",
+            ):
+                nopt = opt[10:]
+                result[nopt] = current[opt]
+            else:
+                result[nopt] = current[opt]
+            _log_default_network(nopt, current[opt])
+
+    for opt in opts:
+        if opt in ("dns", "dns_search") and not isinstance(opts[opt], list):
+            result[opt] = opts[opt].split()
+        else:
+            result[opt] = opts[opt]
+    return result
+
+
+def _raise_error_iface(iface, option, expected):
+    """
+    Log and raise an error with a logical formatted message.
+    """
+    msg = _error_msg_iface(iface, option, expected)
+    log.error(msg)
+    raise AttributeError(msg)
+
+
+def _raise_error_network(option, expected):
+    """
+    Log and raise an error with a logical formatted message.
+    """
+    msg = _error_msg_network(option, expected)
+    log.error(msg)
+    raise AttributeError(msg)
+
+
+def _raise_error_routes(iface, option, expected):
+    """
+    Log and raise an error with a logical formatted message.
+    """
+    msg = _error_msg_routes(iface, option, expected)
+    log.error(msg)
+    raise AttributeError(msg)
+
+
+def _read_file(path):
+    """
+    Reads and returns the contents of a file
+    """
+    try:
+        with salt.utils.files.fopen(path, "rb") as rfh:
+            return _get_non_blank_lines(salt.utils.stringutils.to_unicode(rfh.read()))
+    except OSError:
+        return []  # Return empty list for type consistency
+
+
+def _write_file_iface(iface, data, folder, pattern):
+    """
+    Writes a file to disk
+    """
+    filename = os.path.join(folder, pattern.format(iface))
+    if not os.path.exists(folder):
+        msg = "{} cannot be written. {} does not exist".format(filename, folder)
+        log.error(msg)
+        raise AttributeError(msg)
+    with salt.utils.files.fopen(filename, "w") as fp_:
+        fp_.write(salt.utils.stringutils.to_str(data))
+
+
+def _write_file_network(data, filename):
+    """
+    Writes a file to disk
+    """
+    with salt.utils.files.fopen(filename, "w") as fp_:
+        fp_.write(salt.utils.stringutils.to_str(data))
+
+
+def _get_non_blank_lines(data):
+    lines = data.splitlines()
+    try:  # Discard newlines if they exist
+        lines.remove("")
+    except ValueError:
+        pass
+    return lines
+
+
+def build_interface(iface, iface_type, enabled, **settings):
+    """
+    Build an interface script for a network interface.
```

Comment:
Need to add iface, iface_type, enabled, and settings to the docs.
Also needs a `Returns:` section